map/map_loader/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp

```diff
@@ -69,21 +88,78 @@ Lanelet2MapLoaderNode::Lanelet2MapLoaderNode(const rclcpp::NodeOptions & options
   declare_parameter<std::string>("lanelet2_map_path");
   declare_parameter<double>("center_line_resolution");
   declare_parameter<bool>("use_waypoints");
+  declare_parameter<bool>("enable_differential_map_loading");
+
+  if (get_parameter("enable_differential_map_loading").as_bool()) {
+    differential_loader_module_ = std::make_unique<Lanelet2DifferentialLoaderModule>(
+      this, get_parameter("center_line_resolution").as_double());
+  }
 }
 
 void Lanelet2MapLoaderNode::on_map_projector_info(
   const MapProjectorInfo::Message::ConstSharedPtr msg)
 {
   const auto allow_unsupported_version = get_parameter("allow_unsupported_version").as_bool();
-  const auto lanelet2_filename = get_parameter("lanelet2_map_path").as_string();
+  auto lanelet2_filename = get_parameter("lanelet2_map_path").as_string();
   const auto center_line_resolution = get_parameter("center_line_resolution").as_double();
   const auto use_waypoints = get_parameter("use_waypoints").as_bool();
+  const auto enable_differential_map_loading =
+    get_parameter("enable_differential_map_loading").as_bool();
 
-  // load map from file
-  const auto map = load_map(lanelet2_filename, *msg);
-  if (!map) {
-    RCLCPP_ERROR(get_logger(), "Failed to load lanelet2_map. Not published.");
-    return;
+  lanelet::LaneletMapPtr map;
+
+  if (!enable_differential_map_loading) {
+    map = load_map(lanelet2_filename, *msg);
+    if (!map) {
+      RCLCPP_ERROR(get_logger(), "Failed to load lanelet2_map. Not published.");
+      return;
+    }
+  } else {
```

Comment:
We can publish the whole map regardless of differential map loading.
(or we can add an option to publish the whole map like pointcloud_map_loader)